device_plugin.go

```diff
@@ -223,13 +226,20 @@ func (nedp *NitroEnclavesDevicePlugin) Stop() {
 
 // NewNitroEnclavesDevicePlugin returns an initialized NitroEnclavesDevicePlugin
 func NewNitroEnclavesDevicePlugin() *NitroEnclavesDevicePlugin {
+	// devs slice, determines the pluginapi.ListAndWatchResponse, which lets the kublet know about the available/allocatable "aws.ec2.nitro/nitro_enclaves" devices
+	// in a k8s worker node. Number of devices, in this context does not represent number of "nitro_enclaves" device files present in the host,
+	// instead it can be interpreted as number pods that can share the same host device file. The same host device file "nitro_enclaves",
+	// can be mounted into multiple pods, which can be used to run an enclave.
+	// This lets us to schedule 2 or more pods requiring nitro_enclaves device on the same k8s node/EC2 instance.
```

Comment:
Looks like those 2 or more potentially independent pods will have access to controlling enclaves of each other. Could it be a problem?

Reply:
I think this should not be a problem.

The `nitro_enclaves` device [fops](https://github.com/torvalds/linux/blob/master/drivers/virt/nitro_enclaves/ne_misc_dev.c#L66-L77) only allows the creation of new enclaves through [`ne_ioctl`](https://github.com/torvalds/linux/blob/master/drivers/virt/nitro_enclaves/ne_misc_dev.c#L1738) which will create a new file descriptor for each enclave through [`ne_create_enclave_ioctl`](https://github.com/torvalds/linux/blob/master/drivers/virt/nitro_enclaves/ne_misc_dev.c#L1592). That enclave specific file descriptor allows control of the enclave through [ne_enclave_fops](https://github.com/torvalds/linux/blob/master/drivers/virt/nitro_enclaves/ne_misc_dev.c#L1570-L1576).

So as long as an owning pod does not make an effort to share access to the file descriptor of its enclave with other pods, control to each enclave belongs to the pod that created the enclave.